java/impl/aws/src/main/java/co/worklytics/psoxy/aws/AWSWorkloadIdentityFederationGrantTokenRequestBuilder.java

```diff
@@ -0,0 +1,66 @@
+package co.worklytics.psoxy.aws;
+
+import co.worklytics.psoxy.gateway.ConfigService;
+import co.worklytics.psoxy.gateway.impl.oauth.WorkloadIdentityFederationGrantTokenRequestBuilder;
+import com.google.common.collect.Streams;
+import lombok.NoArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.extern.java.Log;
+import software.amazon.awssdk.services.cognitoidentity.CognitoIdentityClient;
+import software.amazon.awssdk.services.cognitoidentity.model.GetOpenIdTokenForDeveloperIdentityRequest;
+import software.amazon.awssdk.services.cognitoidentity.model.GetOpenIdTokenForDeveloperIdentityResponse;
+
+import javax.inject.Inject;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Federation for AWS using a Cognito instance to have an IDENTITY_ID previously created
+ * identity to obtain its security JWT.
+ * The identity created should match login link DEVELOPER_NAME_ID=CLIENT_ID value created
+ */
+// NOTE: For some reason, this annotation is not working with @NoArgsConstructor(onConstructor_ = @Inject) syntax
+@NoArgsConstructor(onConstructor = @__(@Inject))
```

Comment:
Weird, works in this way but no in the other

Reply:
could be my IDE being flaky

Reply:
Not at all, it happens just running maven outside of the IDE, but no idea why...